joule/providers/aws.py

```diff
@@ -111,7 +114,7 @@ def get_events_from_message_queue(self) -> Iterator[Event]:
         :return: Event
         """
         rx: List[Message] = self._queue.receive_messages(
-            VisibilityTimeout=0, WaitTimeSeconds=1, MaxNumberOfMessages=10
+            VisibilityTimeout=5, WaitTimeSeconds=2, MaxNumberOfMessages=10
```

Comment:
`VisibilityTimeout=5` Has the issue where the message can get picked up by multiple nodes, which we don’t want.

Reply:
Are you certain about this @joedborg? In [1] visibility timeout is defined as "a period of time during which Amazon SQS prevents other consumers from receiving and processing the message" and in [2] it states that "If a message must be received only once, your consumer should delete it within the duration of the visibility timeout". So if we have a visibility timeout of 0 the message is always visible to multiple consumers and there fore can be processed multiple times until one of them deletes it. If we set a visibility timeout of 5 we have 5 seconds to delete a message from the queue without any other consumer having the chance to process it. 5 secs should be enough since the only thing we do is dequeue and delete in https://github.com/canonical/joule/blob/main/joule/providers/aws.py#L107.  

[1] https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-visibility-timeout.html
[2] https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-visibility-timeout.html#configuring-visibility-timeout

Reply:
Oh, I see now why we want visibility to be 0. We actually want all nodes to always have access to all the items of the queue so that a node will process the join message sent to it. If the visibility is not 0 the message intended for a joining node may never reach it because other nodes may keep on reading it and hiding it for the visibility timeout period. Let's take it offline on how we can improve this.

Reply:
It also adds a huge amount of time to stabilising the cluster.

I think it’d be good to split the CF work out of this PR.

Reply:
+1. I reverted the change on the visibility and created a card so we can work on queue handling on another PR.